var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -236,15 +236,19 @@ def mplib_content(spec, pre=None):
     else:
         pre = mpi_spec.prefix
 
+    libname = 'mpi'
+    if 'mpich' in mpi_spec.name:
+        libname = 'mpich'
+
     info = {
         'name':   '{0}-{1}'.format(mpi_spec.name, mpi_spec.version),
         'prefix':  pre,
         'include': inc,
         'bindir':  bin,
         'libdir':  lib,
-        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+        'FLAGS':  '-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX',
         'PINC':   '-I{0}'.format(inc),
-        'PLIBS':  '-L{0} -lmpi'.format(lib),
+        'PLIBS':  '-L{0} -l{1}'.format(lib, libname),
```

Comment:
If you're worried about choosing the correct libs regardless of MPI library, use:
```python
'PLIBS':  mpi_spec.libs.ld_flags,
```
See http://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.LibraryList.ld_flags for more information.

Of course, this won't work depending on `pre`, but I don't know what `pre` is for, so you'll have to decide.

Reply:
I will include this (handling of libmpich) in the next standard update for the openfoam-com package.